docs/website/docs/user-guides/advanced/using-odo-with-other-clusters.md

```diff
@@ -0,0 +1,322 @@
+
+
+ ✗  err: error while streaming command: command terminated with exit code 2
+```
+
+</details>
+
+<details>
+<summary>Or, while running a Go application with <code>go</code> devfile on an Azure Kubernetes Service may end up in an error like this.</summary>
+
+```shell
+$ odo dev
+  __
+ /  \__     Developing using the "places" Devfile
+ \__/  \    Namespace: default
+ /  \__/    odo version: v3.10.0
+ \__/
+
+ ⚠  You are using "default" namespace, odo may not work as expected in the default namespace.
+ ⚠  You may set a new namespace by running `odo create namespace <name>`, or set an existing one by running `odo set namespace <name>`
+
+↪ Running on the cluster in Dev mode
+ •  Waiting for Kubernetes resources  ...
+ ⚠  Pod is Pending
+ ✓  Pod is Running
+ ◐  Syncing files into the container ✗  Command 'tar xf - -C /projects --no-same-owner' in container failed.
+
+ ✗  stdout:
+
+ ✗  stderr: tar: main.go: Cannot open: Permission denied
+tar: .gitignore: Cannot open: Permission denied
+tar: README.md: Cannot open: Permission denied
+tar: devfile.yaml: Cannot open: Permission denied
+tar: go.mod: Cannot open: Permission denied
+tar: Exiting with failure status due to previous errors
+
+
+ ✗  err: error while streaming command: command terminated with exit code 2
+
+ ✗  Syncing files into the container [4s]
+Error occurred on Push - watch command was unable to push component: failed to sync to component with name places: failed to sync to component with name places: unable push files to pod: error while streaming command: command terminated with exit code 2
+
+
+↪ Dev mode
+ Status:
+ Watching for changes in the current directory /tmp/go-app
+
+ Keyboard Commands:
+[Ctrl+c] - Exit and delete resources from the cluster
+     [p] - Manually apply local changes to the application on the cluster
+^CCleaning resources, please wait
+ ✗  Dev mode interrupted by user
+```
+
+</details>
+
+Various factors are responsible for this:
+* Storage Provisioner used for the cluster
+* User set by the container image
+* Location on the container where the files are to be synced
+* Using Ephemeral vs Non-Ephemeral Volumes
+
+Users may encounter storage related permissions issues even while working on a standard Kubernetes or OpenShift cluster.
+
+This guide will discuss some workarounds that can be used to fix these issues.
+
+### Using Ephemeral Volumes
+This is the simplest way to overcome this issue. There are 2 parts to this solution:
+1. Set `odo` preference `Ephemeral` to _true_.
+
+   ```shell
+    odo preference set Ephemeral true -f
+    ```
+2. If the Devfile contains a `volume` component, then set its `ephemeral` property to `true`.
+   The above configuration will use the [`emptyDir` Ephemeral volumes](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) instead of creating Persistent Volumes to mount the source files; it also ensures the current user can read/write to the directories.
```

Comment:
I guess this sentence applies to both 1. and 2., and could be separated from the 2nd bullet point.
I would also suggest linking to this [other page](https://odo.dev/docs/development/architecture/how-odo-works#volumes) in case readers want to learn more about how `odo` works with Volumes?

```suggestion

   The above configuration will use the [`emptyDir` Ephemeral volumes](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) instead of creating Persistent Volumes to mount the source files; it also ensures the current user can read/write to the directories.
```